sample: add readme (#9)
ktaborowski/sdk-sidewalk: samples/template/README.rst

Besides the development kit, a LoRa radio module is also needed for this sample if you build it with the LoRa or FSK configuration.
The supported modules are as follows:

+------------+---------------+------------+
| Radio chip | Board name    | Interface  |
+============+===============+============+
| `SX1262`_  | SX126xMB2xAS  | SPI @ 8Mhz |
+------------+---------------+------------+

.. note::
   The LoRa radio module shield must be connected to the development kit header, and the antenna has to be connected to the radio module.
   For the exact pin assignment, refer to the :ref:`setting_up_hardware_semtech_pinout` section.

.. _template_overview:

Overview
********

You can use this sample as a starting point to implement a Sidewalk device.
The sample shows implementation of the Sidewalk API.

.. _template_ui:

User interface
**************

A button action is triggered when you release the button.
To use a long press action, hold a button for 2 seconds or longer, and release it.

The button assignment is as follows:

   * Button 1 (long press):
      Factory reset - The application informs the Sidewalk stack about the factory reset event.

>       The Sidewalk library clears its configuration from the non-volatile storage.

Reviewer: ```suggestion
      The Sidewalk library clears its configuration from the non-volatile storage and also saved initial link type.
```

>       After a successful reset, the device needs to be registered with the cloud services again.

Reviewer: ```suggestion
      After a successful sample restart, the device needs to be registered with the cloud services again.
```

   * Button 2 (short press):
      In BLE link type - toggle Connection Request - The device requests the Sidewalk Gateway to initiate a connection while the device is advertising through Bluetooth LE.
      After the connection is dropped, the user has to set the beacon state again.
      Gateways may not always be able to process this request, as it depends on the number of devices connected to it.

   * Button 2 (short press):
      In LoRa or FSK link type - get current Device Profiles - The divice profile paramters are printed in logs.

   * Button 2 (long press):
      In LoRa or FSK link type - switch between Device Profiles - The application sets the device profile fields like link, profile ID, window count, sync and async intervals for communication.

   * Button 3:
      Send Hello - This action will queue a message to the cloud.
      If Sidewalk is not ready, it will simply show an error without sending the message.
      The queue will be processed eventually, and all the queued messages will be sent.

   * Button 4 (short press):

>       Change link type. There is a Circular link type change BLE -> FSK -> LoRa -> BLE ...

Reviewer: ```suggestion
      Change link type. There is a Circular link type change BLE -> FSK -> LoRa -> BLE ->. Selected link type is saved and set after sample restart. ``Button 1 (long press)`` reset link type to initial one defined by ``CONFIG_SIDEWALK_APP_LINK_TYPE_(BLE/LORA/FSK)``
```

Note: LoRa link type does not support registration and switching to it on de-registered sample causes error ``Failed to initialize sidewalk err:-6``

Depending on the selected board, a sample supports the following build types:

* ``debug`` -- Debug version of the application - can be used to enable additional features, such as logs or command-line shell, to verify the application behavior.
* ``release`` -- Release version of the application - can be used to enable only the necessary application functionalities to optimize its performance.

You can build the ``release`` firmware for ``build_target`` by running the following command in the project directory:

.. parsed-literal::
   :class: highlight

   $ west build -b *build_target* -- -DCONF_FILE=prj_release.conf

For example:

.. code-block:: console

   $ west build -b nrf52840dk_nrf52840 -- -DCONF_FILE=prj_release.conf

.. note::
    Selecting a build type is optional.
    However, if the build type is not selected, the ``debug`` build type is used by default.

Command Line Interface
======================

The Sidewalk CLI can be set by the Kconfig

CONFIG_SIDEWALK_CLI

The SIDEWALK_CLI enables CLI compatible with sid_dut sample.

To get list of supported CLI commands, you can use `sid help`.

> Power measurement test scenario

Reviewer: ```suggestion
Power measurement application
```

Also Please add **SensorDemoApp application**

===============================

The application may perform the common Sidewalk application steps such us:

#. Registration
#. Time sync
#. Send hello message to cloud
#. Echo messages received from cloud

> This application is especially useful to test release configuration, and can be set by the Kconfig

Reviewer: ```suggestion
This application is especially useful in ``release`` configuration to check power consumption in different Sidewalk states. The states can be monitor by GPIO.
```

CONFIG_SIDEWALK_APP_ECHO

> You can specify the default transport protocol using Kconfig

Reviewer: This akapit is somehow hidden.
In general I would suggest to create two section.
1. Application extensions
    1.1. Power app
    1.2. SensorDemo app
2. Configurations
     2.1. CLI
     2.2. Initial LINK_TYPE
     2.3. Minimal Sidewalk libraries
     2.4. DFU - if it will be selectable

Author: Please see updated version

> CONFIG_SIDEWALK_APP_LINK_TYPE_(BLE/LORA/FSK)

Reviewer: Note: LoRa link type does not support registration and on de-registered sample ``CONFIG_SIDEWALK_APP_LINK_TYPE_LORA`` causes to switch to BLE, registers sample and switch back to LoRa

If a new, correct image is found, the bootloader can only start the application that is placed in the first partition and exchange the application from the second partition.
It may take a few seconds.

Building and running
********************

This sample can be found under :file:`samples/template`.

.. note::
   Before you flash you Sidewalk sample, make sure you completed the following:

      * You downloaded the Sidewalk repository and updated west according to the :ref:`dk_building_sample_app` section.
      * You provisioned your device during the :ref:`setting_up_sidewalk_product`.

   This step needs to be completed only once.
   You do not have to repeat it on every sample rebuild.

To build the sample, follow the steps in the `Building and programming an application`_ documentation.

.. _template_selecting_build:

Selecting a build type
======================

Before you start testing the application, you can select one of the available build types.
To select the build type when building the application from command line, specify it by adding one of the following parameters to the ``west build`` command:

> * Build with the default configuration (LoRa link mode):

Reviewer 2: This was covered by lines 158
